src/pytest_replay/__init__.py

```diff
@@ -106,22 +106,23 @@ def pytest_collection_modifyitems(self, items, config):
 
         with open(replay_file, "r", encoding="UTF-8") as f:
             all_lines = f.readlines()
-            nodeids = {
+            nodeids = dict.fromkeys(
                 json.loads(line)["nodeid"]
                 for line in all_lines
                 if not line.strip().startswith(("#", "//"))
-            }
+            )
+
+        items_dict = {item.nodeid: item for item in items}
         remaining = []
-        deselected = []
-        for item in items:
-            if item.nodeid in nodeids:
+        for nodeid in nodeids:
```

Comment:
Let's add a comment to make this operation more prominent, and add a reference to the original issue for context.

```suggestion
        # Make sure to respect the order from the JSON file (#52).
        for nodeid in nodeids:
```